### ISSUE TITLE
Log no longer available in api

### ISSUE THREAD

**Issue by krl** [ipfs/go-ipfs#1504]
/api/v0/log/tail no longer produces any output.

Should we move this back, or should we change the api somehow?

**Comment by jbenet** [comment]
@whyrusleeping i think the current `/logs` route should be back where `/api/v0/log/tail` was.

**Comment by jbenet** [comment]
@whyrusleeping maybe we should fix this regression now since it's very quick (just moving an http route), and it hasn't been externalized in a release

**Comment by whyrusleeping** [comment]
@jbenet why dont we just move the code that depends on this to point to `/logs` since the log tail is no longer an api command. faking it would break our perfect 'api matches the cli' symmetry.

**Comment by whyrusleeping** [comment]
```
diff --git a/cmd/ipfs/daemon.go b/cmd/ipfs/daemon.go
index 0c31ce0..4201e93 100644
--- a/cmd/ipfs/daemon.go
+++ b/cmd/ipfs/daemon.go
@@ -299,13 +299,13 @@ func serveHTTPApi(req cmds.Request) (error, <-chan error) {
                },
        })
        var opts = []corehttp.ServeOption{
+               corehttp.LogOption(),
                corehttp.CommandsOption(*req.InvocContext()),
                corehttp.WebUIOption,
                apiGw.ServeOption(),
                corehttp.VersionOption(),
                defaultMux("/debug/vars"),
                defaultMux("/debug/pprof/"),
-               corehttp.LogOption(),
                corehttp.PrometheusOption("/debug/metrics/prometheus"),
        }
 
diff --git a/core/corehttp/logs.go b/core/corehttp/logs.go
index 7624644..4fd49de 100644
--- a/core/corehttp/logs.go
+++ b/core/corehttp/logs.go
@@ -37,7 +37,7 @@ func (w *writeErrNotifier) Write(b []byte) (int, error) {
 
 func LogOption() ServeOption {
        return func(n *core.IpfsNode, mux *http.ServeMux) (*http.ServeMux, error) {
-               mux.HandleFunc("/logs", func(w http.ResponseWriter, r *http.Request) {
+               mux.HandleFunc("/api/v0/log/tail", func(w http.ResponseWriter, r *http.Request) {
                        w.WriteHeader(200)
                        wnf, errs := newWriteErrNotifier(w)
                        eventlog.WriterGroup.AddWriter(wnf)

```

**Comment by whyrusleeping** [comment]
Thats all the diff thats required to move it, but i'm not sure we should.

**Comment by jbenet** [comment]
> why dont we just move the code that depends on this to point to /logs since the log tail is no longer an api command. faking it would break our perfect 'api matches the cli' symmetry.

why isn't it a command? users should be able to `ipfs logs tail` still...

**Comment by jbenet** [comment]
- can easily mount the logs at the right location
- continue to support the cli command
- continue to support the existing code out there.

why did we break userland in the first place? is there a reason to?

**Comment by whyrusleeping** [comment]
man, its userland all the way down.

**Comment by whyrusleeping** [comment]
we 'broke' userland because the original motivation was to stop writing logs to disk, and instead change that to be output to an http endpoint, at no point did we think about the log command (which I admit, I forgot existed)

**Comment by jbenet** [comment]
Closed in #1539
